frontend/src/components/Search/useSavedSegments/index.tsx

```diff
@@ -424,33 +363,127 @@ export const useSavedSegments = ({
 			return null
 		}
 
-		const buttonProps =
-			mode === BuilderMode.SEGMENT_UPDATE
-				? {
-						kind: 'primary' as const,
-						emphasis: 'high' as const,
-					}
-				: {
-						kind: 'secondary' as const,
-						emphasis: 'medium' as const,
-					}
+		const buttonProps = [
+			BuilderMode.SEGMENT_NEW,
+			BuilderMode.SEGMENT_UPDATE,
+		].includes(mode)
+			? {
+					kind: 'primary' as const,
+					emphasis: 'high' as const,
+				}
+			: {
+					kind: 'secondary' as const,
+					emphasis: 'medium' as const,
+				}
 
 		return (
-			<Menu>
-				<Menu.Button
-					disabled={segmentsLoading}
-					iconLeft={<IconSolidSegment size={12} />}
-					iconRight={<IconSolidCheveronDown size={12} />}
-					{...buttonProps}
-				>
-					{selectedSegment?.name && (
-						<Text lines="1">{selectedSegment?.name}</Text>
-					)}
-				</Menu.Button>
-				<Menu.List cssClass={styles.menuList}>{menuOptions}</Menu.List>
-			</Menu>
+			<Box
+				display="flex"
+				alignItems="center"
+				gap="0"
+				flexDirection="row"
+				border="secondary"
+				borderRadius="8"
+			>
+				{mode === BuilderMode.SEGMENT_NEW ? (
+					<Menu>
+						<Menu.Button
+							iconLeft={<IconSolidSave size={12} />}
+							{...buttonProps}
+							onClick={(e: any) => {
+								e.preventDefault()
+								setSegmentModalState(SegmentModalState.CREATE)
+							}}
+						>
+							<Text lines="1">Save</Text>
+						</Menu.Button>
```

Comment:
the `save` text gets truncated 

![Screenshot from 2024-11-22 17-09-06](https://github.com/user-attachments/assets/6b18ddf9-4c77-45fb-b555-de6f950b265a)

while the segment text looks right. 
![Screenshot from 2024-11-22 17-16-57](https://github.com/user-attachments/assets/e4ee529f-4433-4a99-a445-4803f088f36e)

could you please fix the save text so that the button is always full-sized to fit the word `Save` ?

Reply:
Ack